From: hmaarrfk <mark.harfouche@conda-forge.org>
Subject: npu_compiler: add missing include guards

These headers have no #pragma once / include guard, so a translation
unit that includes one of them twice (directly or through other headers)
gets "redefinition of ..." errors. This does not show up in a normal
build only because each .cpp happens to include them once; it does as
soon as sources are compiled in unity batches (patch 0016), e.g.
shave_binary_resources.h (ShaveBinaryResources),
inference_execution_simulator.hpp (TaskConfig) and
quantile_float/dialect.hpp (QuantileDialect).

--- a/src/vpux_compiler/include/vpux/compiler/NPU37XX/dialect/IE/IR/ops_interfaces.hpp
+++ b/src/vpux_compiler/include/vpux/compiler/NPU37XX/dialect/IE/IR/ops_interfaces.hpp
@@ -2,6 +2,8 @@
 // Copyright (C) 2023-2026 Intel Corporation
 // SPDX-License-Identifier: Apache-2.0
 //
+
+#pragma once
 
 #include <mlir/IR/Dialect.h>
 
--- a/src/vpux_compiler/include/vpux/compiler/NPU37XX/dialect/VPU/IR/ops_interfaces.hpp
+++ b/src/vpux_compiler/include/vpux/compiler/NPU37XX/dialect/VPU/IR/ops_interfaces.hpp
@@ -2,6 +2,8 @@
 // Copyright (C) 2023-2026 Intel Corporation
 // SPDX-License-Identifier: Apache-2.0
 //
+
+#pragma once
 
 #include <mlir/IR/Dialect.h>
 
--- a/src/vpux_compiler/include/vpux/compiler/NPU37XX/dialect/VPUIP/IR/ops_interfaces.hpp
+++ b/src/vpux_compiler/include/vpux/compiler/NPU37XX/dialect/VPUIP/IR/ops_interfaces.hpp
@@ -2,6 +2,8 @@
 // Copyright (C) 2023-2026 Intel Corporation
 // SPDX-License-Identifier: Apache-2.0
 //
+
+#pragma once
 
 #include <mlir/IR/Dialect.h>
 
--- a/src/vpux_compiler/include/vpux/compiler/NPU37XX/dialect/VPUIPDPU/ops_interfaces.hpp
+++ b/src/vpux_compiler/include/vpux/compiler/NPU37XX/dialect/VPUIPDPU/ops_interfaces.hpp
@@ -2,6 +2,8 @@
 // Copyright (C) 2024-2026 Intel Corporation
 // SPDX-License-Identifier: Apache-2.0
 //
+
+#pragma once
 
 #include "vpux/compiler/dialect/VPUIPDPU/ops.hpp"
 
--- a/src/vpux_compiler/include/vpux/compiler/NPU40XX/dialect/IE/IR/ops_interfaces.hpp
+++ b/src/vpux_compiler/include/vpux/compiler/NPU40XX/dialect/IE/IR/ops_interfaces.hpp
@@ -2,6 +2,8 @@
 // Copyright (C) 2025-2026 Intel Corporation
 // SPDX-License-Identifier: Apache-2.0
 //
+
+#pragma once
 
 #include <mlir/IR/Dialect.h>
 
--- a/src/vpux_compiler/include/vpux/compiler/NPU40XX/dialect/VPU/IR/ops_interfaces.hpp
+++ b/src/vpux_compiler/include/vpux/compiler/NPU40XX/dialect/VPU/IR/ops_interfaces.hpp
@@ -2,6 +2,8 @@
 // Copyright (C) 2024-2026 Intel Corporation
 // SPDX-License-Identifier: Apache-2.0
 //
+
+#pragma once
 
 #include <mlir/IR/Dialect.h>
 
--- a/src/vpux_compiler/include/vpux/compiler/NPU40XX/dialect/VPUIP/IR/ops_interfaces.hpp
+++ b/src/vpux_compiler/include/vpux/compiler/NPU40XX/dialect/VPUIP/IR/ops_interfaces.hpp
@@ -2,6 +2,8 @@
 // Copyright (C) 2024-2026 Intel Corporation
 // SPDX-License-Identifier: Apache-2.0
 //
+
+#pragma once
 
 #include <mlir/IR/Dialect.h>
 
--- a/src/vpux_compiler/include/vpux/compiler/NPU40XX/dialect/VPUIPDPU/lower_to_registers.hpp
+++ b/src/vpux_compiler/include/vpux/compiler/NPU40XX/dialect/VPUIPDPU/lower_to_registers.hpp
@@ -2,6 +2,8 @@
 // Copyright (C) 2024-2026 Intel Corporation
 // SPDX-License-Identifier: Apache-2.0
 //
+
+#pragma once
 
 #include "vpux/compiler/NPU40XX/dialect/NPUReg40XX/descriptors.hpp"
 #include "vpux/compiler/NPU40XX/dialect/NPUReg40XX/types.hpp"
--- a/src/vpux_compiler/include/vpux/compiler/NPU40XX/dialect/VPUIPDPU/ops_interfaces.hpp
+++ b/src/vpux_compiler/include/vpux/compiler/NPU40XX/dialect/VPUIPDPU/ops_interfaces.hpp
@@ -2,6 +2,8 @@
 // Copyright (C) 2024-2026 Intel Corporation
 // SPDX-License-Identifier: Apache-2.0
 //
+
+#pragma once
 
 #include "vpux/compiler/dialect/VPUIPDPU/ops.hpp"
 
--- a/src/vpux_compiler/include/vpux/compiler/NPU40XX/dialect/VPUIPDPU/transforms/passes/expand_dpu_config/expand_dpu_config_invariant_ppe.hpp
+++ b/src/vpux_compiler/include/vpux/compiler/NPU40XX/dialect/VPUIPDPU/transforms/passes/expand_dpu_config/expand_dpu_config_invariant_ppe.hpp
@@ -2,6 +2,8 @@
 // Copyright (C) 2024-2026 Intel Corporation
 // SPDX-License-Identifier: Apache-2.0
 //
+
+#pragma once
 
 #include "vpux/compiler/dialect/VPU/IR/attributes.hpp"
 #include "vpux/compiler/dialect/VPUIPDPU/attributes.hpp"
--- a/src/vpux_compiler/include/vpux/compiler/NPU40XX/dialect/VPUIPDPU/transforms/passes/expand_dpu_config/expand_dpu_config_variant.hpp
+++ b/src/vpux_compiler/include/vpux/compiler/NPU40XX/dialect/VPUIPDPU/transforms/passes/expand_dpu_config/expand_dpu_config_variant.hpp
@@ -2,6 +2,8 @@
 // Copyright (C) 2024-2026 Intel Corporation
 // SPDX-License-Identifier: Apache-2.0
 //
+
+#pragma once
 
 #include "vpux/compiler/NPU40XX/dialect/VPUIPDPU/ops.hpp"
 #include "vpux/compiler/dialect/VPUASM/ops.hpp"
--- a/src/vpux_compiler/include/vpux/compiler/NPU40XX/dialect/VPUIPDPU/transforms/passes/expand_dpu_config/expand_dpu_config_variant_general.hpp
+++ b/src/vpux_compiler/include/vpux/compiler/NPU40XX/dialect/VPUIPDPU/transforms/passes/expand_dpu_config/expand_dpu_config_variant_general.hpp
@@ -2,6 +2,8 @@
 // Copyright (C) 2024-2026 Intel Corporation
 // SPDX-License-Identifier: Apache-2.0
 //
+
+#pragma once
 
 #include "vpux/compiler/NPU40XX/dialect/VPUIPDPU/ops.hpp"
 #include "vpux/compiler/dialect/VPUASM/ops.hpp"
--- a/src/vpux_compiler/include/vpux/compiler/NPU40XX/dialect/VPUIPDPU/transforms/passes/expand_dpu_config/expand_dpu_config_variant_idu.hpp
+++ b/src/vpux_compiler/include/vpux/compiler/NPU40XX/dialect/VPUIPDPU/transforms/passes/expand_dpu_config/expand_dpu_config_variant_idu.hpp
@@ -2,6 +2,8 @@
 // Copyright (C) 2024-2026 Intel Corporation
 // SPDX-License-Identifier: Apache-2.0
 //
+
+#pragma once
 
 #include "vpux/compiler/NPU40XX/dialect/VPUIPDPU/ops.hpp"
 #include "vpux/compiler/dialect/VPUASM/ops.hpp"
--- a/src/vpux_compiler/include/vpux/compiler/NPU40XX/dialect/VPUIPDPU/transforms/passes/expand_dpu_config/expand_dpu_config_variant_odu.hpp
+++ b/src/vpux_compiler/include/vpux/compiler/NPU40XX/dialect/VPUIPDPU/transforms/passes/expand_dpu_config/expand_dpu_config_variant_odu.hpp
@@ -2,6 +2,8 @@
 // Copyright (C) 2024-2026 Intel Corporation
 // SPDX-License-Identifier: Apache-2.0
 //
+
+#pragma once
 
 #include "vpux/compiler/NPU40XX/dialect/VPUIPDPU/ops.hpp"
 #include "vpux/compiler/dialect/VPUASM/ops.hpp"
--- a/src/vpux_compiler/include/vpux/compiler/NPU50XX/dialect/IE/IR/ops_interfaces.hpp
+++ b/src/vpux_compiler/include/vpux/compiler/NPU50XX/dialect/IE/IR/ops_interfaces.hpp
@@ -2,6 +2,8 @@
 // Copyright (C) 2026 Intel Corporation
 // SPDX-License-Identifier: Apache-2.0
 //
+
+#pragma once
 
 #include <mlir/IR/Dialect.h>
 
--- a/src/vpux_compiler/include/vpux/compiler/NPU50XX/dialect/VPU/IR/ops_interfaces.hpp
+++ b/src/vpux_compiler/include/vpux/compiler/NPU50XX/dialect/VPU/IR/ops_interfaces.hpp
@@ -2,6 +2,8 @@
 // Copyright (C) 2024-2026 Intel Corporation
 // SPDX-License-Identifier: Apache-2.0
 //
+
+#pragma once
 
 #include <mlir/IR/Dialect.h>
 
--- a/src/vpux_compiler/include/vpux/compiler/NPU50XX/dialect/VPUIP/IR/ops_interfaces.hpp
+++ b/src/vpux_compiler/include/vpux/compiler/NPU50XX/dialect/VPUIP/IR/ops_interfaces.hpp
@@ -3,6 +3,8 @@
 // SPDX-License-Identifier: Apache-2.0
 //
 
+#pragma once
+
 #include <mlir/IR/Dialect.h>
 
 namespace vpux::VPUIP::arch50xx {}  // namespace vpux::VPUIP::arch50xx
--- a/src/vpux_compiler/include/vpux/compiler/NPU50XX/dialect/VPUIPDPU/lower_to_registers.hpp
+++ b/src/vpux_compiler/include/vpux/compiler/NPU50XX/dialect/VPUIPDPU/lower_to_registers.hpp
@@ -2,6 +2,8 @@
 // Copyright (C) 2024-2026 Intel Corporation
 // SPDX-License-Identifier: Apache-2.0
 //
+
+#pragma once
 
 #include "vpux/compiler/NPU40XX/dialect/VPUIPDPU/lower_to_registers.hpp"
 #include "vpux/compiler/NPU50XX/dialect/NPUReg50XX/descriptors.hpp"
--- a/src/vpux_compiler/include/vpux/compiler/NPU50XX/dialect/VPUIPDPU/ops_interfaces.hpp
+++ b/src/vpux_compiler/include/vpux/compiler/NPU50XX/dialect/VPUIPDPU/ops_interfaces.hpp
@@ -2,6 +2,8 @@
 // Copyright (C) 2024-2026 Intel Corporation
 // SPDX-License-Identifier: Apache-2.0
 //
+
+#pragma once
 
 #include <mlir/IR/Dialect.h>
 
--- a/src/vpux_compiler/include/vpux/compiler/NPU50XX/dialect/VPUIPDPU/transforms/passes/expand_dpu_config/expand_dpu_config_invariant.hpp
+++ b/src/vpux_compiler/include/vpux/compiler/NPU50XX/dialect/VPUIPDPU/transforms/passes/expand_dpu_config/expand_dpu_config_invariant.hpp
@@ -2,6 +2,8 @@
 // Copyright (C) 2024-2026 Intel Corporation
 // SPDX-License-Identifier: Apache-2.0
 //
+
+#pragma once
 
 #include "vpux/compiler/NPU40XX/dialect/VPUIPDPU/transforms/passes/expand_dpu_config/expand_dpu_config_invariant.hpp"
 #include "vpux/compiler/NPU50XX/dialect/VPUIPDPU/ops.hpp"
--- a/src/vpux_compiler/include/vpux/compiler/NPU50XX/dialect/VPUIPDPU/transforms/passes/expand_dpu_config/expand_dpu_config_variant.hpp
+++ b/src/vpux_compiler/include/vpux/compiler/NPU50XX/dialect/VPUIPDPU/transforms/passes/expand_dpu_config/expand_dpu_config_variant.hpp
@@ -2,6 +2,8 @@
 // Copyright (C) 2024-2026 Intel Corporation
 // SPDX-License-Identifier: Apache-2.0
 //
+
+#pragma once
 
 #include "vpux/compiler/NPU40XX/dialect/VPUIPDPU/transforms/passes/expand_dpu_config/expand_dpu_config_variant.hpp"
 #include "vpux/compiler/NPU50XX/dialect/VPUIPDPU/ops.hpp"
--- a/src/vpux_compiler/include/vpux/compiler/NPU50XX/dialect/VPUIPDPU/transforms/passes/expand_dpu_config/expand_dpu_config_variant_idu.hpp
+++ b/src/vpux_compiler/include/vpux/compiler/NPU50XX/dialect/VPUIPDPU/transforms/passes/expand_dpu_config/expand_dpu_config_variant_idu.hpp
@@ -2,6 +2,8 @@
 // Copyright (C) 2024-2026 Intel Corporation
 // SPDX-License-Identifier: Apache-2.0
 //
+
+#pragma once
 
 #include "vpux/compiler/NPU40XX/dialect/VPUIPDPU/transforms/passes/expand_dpu_config/expand_dpu_config_variant_idu.hpp"
 #include "vpux/compiler/NPU50XX/dialect/VPUIPDPU/ops.hpp"
--- a/src/vpux_compiler/include/vpux/compiler/act_kernels/shave_binary_resources.h
+++ b/src/vpux_compiler/include/vpux/compiler/act_kernels/shave_binary_resources.h
@@ -2,6 +2,8 @@
 // Copyright (C) 2022-2026 Intel Corporation
 // SPDX-License-Identifier: Apache-2.0
 //
+
+#pragma once
 
 #include <llvm/ADT/ArrayRef.h>
 #include <llvm/ADT/StringRef.h>
--- a/src/vpux_compiler/include/vpux/compiler/conversion/passes/VPU2VPUIP/bufferize_sw_ops_interface.hpp
+++ b/src/vpux_compiler/include/vpux/compiler/conversion/passes/VPU2VPUIP/bufferize_sw_ops_interface.hpp
@@ -2,6 +2,8 @@
 // Copyright (C) 2024-2026 Intel Corporation
 // SPDX-License-Identifier: Apache-2.0
 //
+
+#pragma once
 
 #include "vpux/compiler/conversion/passes/VPU2VPUIP/bufferizable_ops_interface.hpp"
 #include "vpux/compiler/dialect/Shave/IR/dialect.hpp"
--- a/src/vpux_compiler/include/vpux/compiler/conversion/passes/VPU2VPUIP/bufferize_vpu_nce_ops_interface.hpp
+++ b/src/vpux_compiler/include/vpux/compiler/conversion/passes/VPU2VPUIP/bufferize_vpu_nce_ops_interface.hpp
@@ -3,6 +3,8 @@
 // SPDX-License-Identifier: Apache-2.0
 //
 
+#pragma once
+
 #include "vpux/utils/logger/logger.hpp"
 
 #include <mlir/Dialect/Bufferization/Transforms/OneShotAnalysis.h>
--- a/src/vpux_compiler/include/vpux/compiler/core/types/quantile_float/dialect.hpp
+++ b/src/vpux_compiler/include/vpux/compiler/core/types/quantile_float/dialect.hpp
@@ -2,6 +2,8 @@
 // Copyright (C) 2024-2026 Intel Corporation
 // SPDX-License-Identifier: Apache-2.0
 //
+
+#pragma once
 
 #include <llvm/ADT/TypeSwitch.h>
 #include <mlir/IR/DialectImplementation.h>
--- a/src/vpux_compiler/include/vpux/compiler/dialect/IE/interfaces/common_rewriters/convert_to_palletization_lut.hpp
+++ b/src/vpux_compiler/include/vpux/compiler/dialect/IE/interfaces/common_rewriters/convert_to_palletization_lut.hpp
@@ -2,6 +2,8 @@
 // Copyright (C) 2025-2026 Intel Corporation
 // SPDX-License-Identifier: Apache-2.0
 //
+
+#pragma once
 
 #include "vpux/compiler/dialect/IE/IR/ops/data_type.hpp"
 #include "vpux/compiler/utils/rewriter.hpp"
--- a/src/vpux_compiler/include/vpux/compiler/dialect/IE/interfaces/common_rewriters/fuse_outstanding_quant.hpp
+++ b/src/vpux_compiler/include/vpux/compiler/dialect/IE/interfaces/common_rewriters/fuse_outstanding_quant.hpp
@@ -2,6 +2,8 @@
 // Copyright (C) 2024-2026 Intel Corporation
 // SPDX-License-Identifier: Apache-2.0
 //
+
+#pragma once
 
 #include "vpux/compiler/dialect/IE/IR/ops/pooling.hpp"
 #include "vpux/utils/core/array_ref.hpp"
--- a/src/vpux_compiler/include/vpux/compiler/dialect/IE/interfaces/common_rewriters/fuse_quantized_ops.hpp
+++ b/src/vpux_compiler/include/vpux/compiler/dialect/IE/interfaces/common_rewriters/fuse_quantized_ops.hpp
@@ -2,6 +2,8 @@
 // Copyright (C) 2024-2026 Intel Corporation
 // SPDX-License-Identifier: Apache-2.0
 //
+
+#pragma once
 
 #include "vpux/compiler/dialect/IE/IR/ops/activation.hpp"
 #include "vpux/compiler/dialect/IE/IR/ops/convolution.hpp"
--- a/src/vpux_compiler/include/vpux/compiler/dialect/IE/transforms/passes/convert_to_mixed_precision.hpp
+++ b/src/vpux_compiler/include/vpux/compiler/dialect/IE/transforms/passes/convert_to_mixed_precision.hpp
@@ -2,6 +2,8 @@
 // Copyright (C) 2022-2026 Intel Corporation
 // SPDX-License-Identifier: Apache-2.0
 //
+
+#pragma once
 
 #include "vpux/compiler/core/types/quantile_float/types.hpp"
 #include "vpux/compiler/dialect/IE/IR/ops/activation.hpp"
--- a/src/vpux_compiler/include/vpux/compiler/dialect/IE/transforms/rewriters/expand_with_layer_rewriter.hpp
+++ b/src/vpux_compiler/include/vpux/compiler/dialect/IE/transforms/rewriters/expand_with_layer_rewriter.hpp
@@ -2,6 +2,8 @@
 // Copyright (C) 2024-2026 Intel Corporation
 // SPDX-License-Identifier: Apache-2.0
 //
+
+#pragma once
 
 #include "vpux/compiler/dialect/IE/IR/ops/data_movement.hpp"
 
--- a/src/vpux_compiler/include/vpux/compiler/dialect/IE/transforms/rewriters/propagate_transpose_affine_reshape_common.hpp
+++ b/src/vpux_compiler/include/vpux/compiler/dialect/IE/transforms/rewriters/propagate_transpose_affine_reshape_common.hpp
@@ -2,6 +2,8 @@
 // Copyright (C) 2024-2026 Intel Corporation
 // SPDX-License-Identifier: Apache-2.0
 //
+
+#pragma once
 
 #include "vpux/compiler/dialect/IE/IR/ops/activation.hpp"
 #include "vpux/compiler/dialect/IE/IR/ops/data_movement.hpp"
--- a/src/vpux_compiler/include/vpux/compiler/dialect/VPU/interfaces/common_utils/layer_permute_ie.hpp
+++ b/src/vpux_compiler/include/vpux/compiler/dialect/VPU/interfaces/common_utils/layer_permute_ie.hpp
@@ -2,6 +2,8 @@
 // Copyright (C) 2025-2026 Intel Corporation
 // SPDX-License-Identifier: Apache-2.0
 //
+
+#pragma once
 
 #include "vpux/compiler/core/attributes/dims_order.hpp"
 
--- a/src/vpux_compiler/include/vpux/compiler/dialect/VPU/utils/json_utils.hpp
+++ b/src/vpux_compiler/include/vpux/compiler/dialect/VPU/utils/json_utils.hpp
@@ -2,6 +2,8 @@
 // Copyright (C) 2022-2025 Intel Corporation
 // SPDX-License-Identifier: Apache-2.0
 //
+
+#pragma once
 
 #include "vpux/compiler/dialect/VPU/utils/manual_strategy_utils.hpp"
 #include "vpux/utils/core/dense_map.hpp"
--- a/src/vpux_compiler/include/vpux/compiler/dialect/VPU/utils/sw_tiling_interface_utils.hpp
+++ b/src/vpux_compiler/include/vpux/compiler/dialect/VPU/utils/sw_tiling_interface_utils.hpp
@@ -2,6 +2,8 @@
 // Copyright (C) 2025-2026 Intel Corporation
 // SPDX-License-Identifier: Apache-2.0
 //
+
+#pragma once
 
 #include "vpux/compiler/dialect/VPU/IR/dialect.hpp"
 #include "vpux/compiler/dialect/VPU/IR/ops_interfaces.hpp"
--- a/src/vpux_compiler/include/vpux/compiler/dialect/VPUIPDPU/ops_fwd.hpp
+++ b/src/vpux_compiler/include/vpux/compiler/dialect/VPUIPDPU/ops_fwd.hpp
@@ -3,4 +3,6 @@
 // SPDX-License-Identifier: Apache-2.0
 //
 
+#pragma once
+
 #include <vpux/compiler/dialect/VPUIPDPU/ops.hpp.inc>
--- a/src/vpux_compiler/include/vpux/compiler/dialect/VPUMI37XX/kernel_params_utils.hpp
+++ b/src/vpux_compiler/include/vpux/compiler/dialect/VPUMI37XX/kernel_params_utils.hpp
@@ -2,6 +2,8 @@
 // Copyright (C) 2022-2026 Intel Corporation
 // SPDX-License-Identifier: Apache-2.0
 //
+
+#pragma once
 
 #include "vpux/compiler/dialect/VPUIP/IR/ops.hpp"
 
--- a/src/vpux_compiler/include/vpux/compiler/dialect/VPUMI37XX/utils.hpp
+++ b/src/vpux_compiler/include/vpux/compiler/dialect/VPUMI37XX/utils.hpp
@@ -2,6 +2,8 @@
 // Copyright (C) 2022-2026 Intel Corporation
 // SPDX-License-Identifier: Apache-2.0
 //
+
+#pragma once
 
 #include "vpux/compiler/dialect/VPUMI37XX/ops.hpp"
 
--- a/src/vpux_compiler/include/vpux/compiler/dialect/VPUMI40XX/wlm_utils.hpp
+++ b/src/vpux_compiler/include/vpux/compiler/dialect/VPUMI40XX/wlm_utils.hpp
@@ -2,6 +2,8 @@
 // Copyright (C) 2024-2026 Intel Corporation
 // SPDX-License-Identifier: Apache-2.0
 //
+
+#pragma once
 
 #include "vpux/compiler/dialect/VPUMI40XX/ops.hpp"
 #include "vpux/compiler/dialect/VPUMI40XX/ops_interfaces.hpp"
--- a/src/vpux_compiler/include/vpux/compiler/dialect/VPURT/interfaces/inference_execution_simulator.hpp
+++ b/src/vpux_compiler/include/vpux/compiler/dialect/VPURT/interfaces/inference_execution_simulator.hpp
@@ -2,6 +2,8 @@
 // Copyright (C) 2023-2026 Intel Corporation
 // SPDX-License-Identifier: Apache-2.0
 //
+
+#pragma once
 
 #include "vpux/compiler/core/cycle_cost_info.hpp"
 #include "vpux/compiler/dialect/VPURT/IR/task.hpp"
--- a/src/vpux_compiler/include/vpux/compiler/dynamic_rewriter/passes.hpp
+++ b/src/vpux_compiler/include/vpux/compiler/dynamic_rewriter/passes.hpp
@@ -2,6 +2,8 @@
 // Copyright (C) 2025-2026 Intel Corporation
 // SPDX-License-Identifier: Apache-2.0
 //
+
+#pragma once
 
 #include "vpux/compiler/dynamic_rewriter/dynamic_rewriter_factory.hpp"
 
--- a/src/vpux_compiler/src/bitc/include/commons.hpp
+++ b/src/vpux_compiler/src/bitc/include/commons.hpp
@@ -2,6 +2,8 @@
 // Copyright (C) 2025-2026 Intel Corporation
 // SPDX-License-Identifier: Apache-2.0
 //
+
+#pragma once
 
 #include <cstdint>
 #include "BitStream.hpp"
--- a/src/vpux_utils/include/vpux/utils/profiling/utils.hpp
+++ b/src/vpux_utils/include/vpux/utils/profiling/utils.hpp
@@ -2,6 +2,8 @@
 // Copyright (C) 2025-2026 Intel Corporation
 // SPDX-License-Identifier: Apache-2.0
 //
+
+#pragma once
 
 #include <cstdint>
 #include <string>
